numpy/f2py/tests/test_callback.py

```diff
@@ -97,24 +97,27 @@ def callback(code):
         r = f(callback)
         assert r == 0
 
-    @pytest.mark.skipif(sys.platform == "win32",
-                        reason="Fails with MinGW64 Gfortran (Issue #9673)")
+    @pytest.mark.skipif(sys.platform == 'win32',
+                        reason='Fails with MinGW64 Gfortran (Issue #9673)')
     def test_string_callback_array(self):
         # See gh-10027
-        cu = np.zeros((1, 8), "S1")
+        cu1 = np.zeros((1, ), "S8")
+        cu2 = np.zeros((1, 8), "c")
```

Comment:
This particular case causes the abort in the tests.

Reply:
Actually this is expected, since the shape of `cu2` is `(1,8)` which means `callback` will return `1`. Additionally the `dtype` is `S1` instead of `S8` so it would return `2` even if the shape is fixed.

Reply:
This is because `len(cu2)==1` which means the incorrect extent is declared for the Fortran code `character*8 cu(lencu)`. `cu2.size` should be used instead.